src/components/Pagination/Pagination.tsx

```diff
@@ -1 +1,86 @@
-export const Pagination = () => {};
+import classNames from 'classnames';
+import { MouseEventHandler } from 'react';
+import { getNumbers } from '../../utils';
+
+type Props = {
+  total: number;
+  perPage: number;
+  currentPage: number;
+  onPageChange: (page: number) => void;
+  onPreviousPage: MouseEventHandler<HTMLAnchorElement>
+  onNextPage: MouseEventHandler<HTMLAnchorElement>
+};
+
+export const Pagination: React.FC<Props> = ({
+  total,
+  perPage,
+  currentPage,
+  onPageChange,
+  onPreviousPage,
+  onNextPage,
+}) => {
+  const totalPages = Math.ceil(total / perPage);
+
+  const handleSetPage = (event: React.MouseEvent<HTMLAnchorElement>) => {
+    event.preventDefault();
+
+    const page = Number(event.currentTarget.dataset.currentPage);
+
+    if (page === currentPage) {
+      return;
+    }
+
+    onPageChange(page);
+  };
+
+  return (
+    <ul className="pagination">
+      <li
+        className={classNames(
+          'page-item', { disabled: currentPage === 1 },
+        )}
+      >
+        <a
+          data-cy="prevLink"
+          className="page-link"
+          href="#prev"
+          aria-disabled={currentPage === 1}
+          onClick={onPreviousPage}
+        >
+          «
+        </a>
+      </li>
+      {getNumbers(1, totalPages).map((page) => (
+        <li className={classNames(
+          'page-item', { active: page === currentPage },
+        )}
+        >
+          <a
+            data-cy="pageLink"
+            className="page-link"
+            data-current-page={page}
+            href={`#${page}`}
+            onClick={handleSetPage}
+          >
+            {page}
+          </a>
+        </li>
+      ))}
+      <li
+        className={classNames(
+          'page-item', { disabled: currentPage === totalPages },
+        )}
+      >
+        <a
+          data-cy="nextLink"
+          className="page-link"
+          href="#next"
+          aria-disabled={currentPage === totalPages}
+          onClick={onNextPage}
```

Comment:
```suggestion
          onClick={handleNextPage} // add new handler inside current component
```

You don't need this handler, cause you have `currentPage` value in props and `onPageChange` also. You can use only them to pass new value to parent component.